proteus/mprans/RANS3PF.py

```diff
@@ -630,25 +629,29 @@ def initializeMesh(self, mesh):
     def initializeElementQuadrature(self, t, cq):
         # VRANS
         self.q_phi_solid = numpy.ones(cq[('u', 0)].shape, 'd')
-        self.q_velocity_solid = numpy.zeros(cq[('velocity', 0)].shape, 'd')
-        self.q_porosity = numpy.ones(cq[('u', 0)].shape, 'd')
         self.q_dragAlpha = numpy.ones(cq[('u', 0)].shape, 'd')
         self.q_dragAlpha.fill(self.dragAlpha)
         self.q_dragBeta = numpy.ones(cq[('u', 0)].shape, 'd')
         self.q_dragBeta.fill(self.dragBeta)
         if self.setParamsFunc is not None:
             self.setParamsFunc(
                 cq['x'],
-                self.q_porosity,
+                self.q_vos,
                 self.q_dragAlpha,
                 self.q_dragBeta)
+            #convert from porosity to volume of sediment
+            self.q_vos -= 1.0
+            self.q_vos *= -1.0
         else:
             # TODO make loops faster
             if self.porosityTypes is not None:
-                for eN in range(self.q_porosity.shape[0]):
-                    self.q_porosity[
+                for eN in range(self.q_vos.shape[0]):
+                    self.q_vos[
```

Comment:
I guess this addition is for settting vos from the porosityTypes variable (i.e. from the region basically)